Examples/Run/Reconstruction/Common/CMakeLists.txt

```
@@ -5,18 +5,34 @@ add_library(
 target_include_directories(
   ActsExamplesRecTracksCommon
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
-target_link_libraries(
-  ActsExamplesRecTracksCommon
-  PUBLIC
-    ActsExamplesFramework
-    ActsExamplesCommon
-    ActsExamplesDigitization
-    ActsExamplesTrackFitting
-    ActsExamplesTrackFinding
-    ActsExamplesMagneticField
-    ActsExamplesTruthTracking
-    ActsExamplesIoCsv
-    ActsExamplesIoPerformance)    
+if(ACTS_BUILD_PLUGIN_ONNX)
+  target_link_libraries(
+    ActsExamplesRecTracksCommon
+    PUBLIC
+      ActsExamplesFramework
+      ActsExamplesCommon
+      ActsExamplesDigitization
+      ActsExamplesTrackFitting
+      ActsExamplesTrackFinding
+      ActsExamplesMagneticField
+      ActsExamplesTruthTracking
+      ActsExamplesIoCsv
+      ActsExamplesIoPerformance
+      ActsPluginOnnx)
+else()
+  target_link_libraries(
+    ActsExamplesRecTracksCommon
+    PUBLIC
+      ActsExamplesFramework
+      ActsExamplesCommon
+      ActsExamplesDigitization
+      ActsExamplesTrackFitting
+      ActsExamplesTrackFinding
+      ActsExamplesMagneticField
+      ActsExamplesTruthTracking
+      ActsExamplesIoCsv
+      ActsExamplesIoPerformance)
+endif()
```

You can call `target_link_libraries` multiple times. A simple way would therefore be to only add a `if`-clause for the onnx runtime, i.e.
```suggestion
target_link_libraries(
  ActsExamplesRecTracksCommon
  PUBLIC
    ActsExamplesFramework
    ActsExamplesCommon
    ActsExamplesDigitization
    ActsExamplesTrackFitting
    ActsExamplesTrackFinding
    ActsExamplesMagneticField
    ActsExamplesTruthTracking
    ActsExamplesIoCsv
    ActsExamplesIoPerformance)

if(ACTS_BUILD_PLUGIN_ONNX)
  target_compile_definitions(
    ActsExamplesRecTracksCommon PUBLIC ACTS_PLUGIN_ONNX)
  target_link_libraries(
    ActsExamplesRecTracksCommon PUBLIC ActsPluginOnnx)
else()
```
where I also added the per-target compile definition.